content/post/policy-based-design-cpp.md

```diff
@@ -0,0 +1,179 @@
++++
+
+title = "Policy based design in C++"
+author = "vmrob"
+draft = false
+date = "2015-09-01T19:05:13-06:00"
+tags = ["C++"]
+description = "Impementing the strategy pattern through the use of templatized mixins in C++"
+
++++
+
+### Design Patterns
+
+As software developers, we use design patterns to name and classify common programming idioms using a common set of jargon. It allows us to describe the higher level design of our code while providing a good idea of what a set of code accomplishes without having to understand everything about it.
```

Comment:
The title and first sentence or two often decide whether or not this thing actually gets read or not. Might want to consider a more to-the-point or thought provoking hook.